src/DataObjects/Request.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Dredd\DataObjects;
+
+class Request
+{
+    /** @var string $body */
+    public $body;
+
+    /**
+     * Can be manually set in {@link https://dredd.org/en/latest/hooks/index.html#hooks hooks}
+     *  - `utf-8` (string) - indicates `body` contains a textual content encoded in UTF-8
+     *  - `base64` (string) - indicates `body` contains a binary content encoded in Base64
+     *
+     * @var string $bodyEncoding
+     * @psalm-var 'utf-8'|'base64' $bodyEncoding
+     */
+    public $bodyEncoding;
+
+    /**
+     * Keys are HTTP header names, values are HTTP header contents
+     * @var object $headers
```

Comment:
This is an array of objects so I think this should be `@var array $headers`

Reply:
According to the docs, it's an object. Also when working with it, it was an object as well.

https://dredd.org/en/latest/data-structures.html#transaction-object

Take a look at request, it's an object there.

It may be that it doesn't matter, and it can be an array, and if thats the case, I think it would be infinitely easier to work with keyed arrays than objects.

```php
$header = "Content-Type";
$transaction->request->$header = "application/json";
```

That's something I have had to do recently, and it's an awful way of working with headers, so hopefully you are right. I'll give it a test.

Tbf, I think when it's json encoded it'll be the same, the problem is when it's json decoded... Though I suppose in this case, we can cast in the end, so shouldn't matter.